src/test/suite/extensionFunctions.test.ts

```diff
@@ -41,4 +43,24 @@ suite('Extension Functions Test Suite', () => {
 		assert.notEqual(terminalC, undefined);
 	});
 
+	test('open new terminal and then close it', async function() {
+		const terminalNameToClose = 'terminalToKill';
+		var terminal : vscode.Terminal | undefined = extensionFunctions.findTerminal(terminalNameToClose);
+		assert.equal(terminal, undefined);
+
+		// if it can't find the terminal, it will error out
+		await extensionFunctions.closeTerminal(terminalNameToClose).catch( (error) => {
+			assert.notEqual(error, undefined);
+		});
+
+		await extensionFunctions.startTerminal(terminalNameToClose);
+		terminal = extensionFunctions.findTerminal(terminalNameToClose);
+		assert.notEqual(terminal, undefined);
+
+		await extensionFunctions.closeTerminal(terminalNameToClose).finally( async () => {
+			terminal = extensionFunctions.findTerminal(terminalNameToClose);
+			// the terminal should be disposed at this point, but we can't test for it
```

Comment:
I have not found a good way to test to see if the terminal has been disposed, but looking at it in the debugger at this point that is the case. Open to suggestions if anybody has any.